### PR TITLE
Break out conda management

### PR DESCRIPTION
**Pull request by jomey**
Create a dedicated folder to manage the conda environment and associated lock files.
Beefed up the `lock-environment.sh` file a bit and added some checks.

### CODE REVIEW

**Comment by jomey** [comment]
I haven't run the script to generate new lock files. There are quite a few more changes other than adding the `cookiecutter`
Want to create a separate PR to better review them and be able to revert.
Then I would also need someone with a Mac to test the .lock file for that platform

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -23,13 +23,17 @@ Ready to contribute? Here's a quick guide
     git checkout -b name-of-your-bugfix-or-feature
     ```
 
-1. Create and activate the "hackweek" conda environment. NOTE: if you're running linux or windows use `binder/conda-linux-64.lock`:
+1. Create and activate the "hackweek" conda environment. 
+
+   __NOTE__: If you're running linux or Windows use `binder/conda-linux-64.lock`
```
Comment (JessicaS11) [inline CONTRIBUTING.md:28]:
```suggestion
   __NOTE__: If you're running linux or Windows use `conda/conda-linux-64.lock`
```

**Review by JessicaS11 — CHANGES_REQUESTED** [review]
See comment - otherwise looks good to me!

**Comment by jomey** [comment]
Going to wait until PR #44 is in to test this against the updated GH Actions

**Review by JessicaS11 — APPROVED** [review]

**Comment by jomey** [comment]
Checks for test are failing and include this error:
```
./scripts/deploy_website.sh: line 24: cookiecutter: command not found
```
As mentioned above in a comment, I will follow up with a separate PR to update the lock files and add cookiecutter.

**Comment by scottyhq** [comment]
> As mentioned above in a comment, I will follow up with a separate PR to update the lock files and add cookiecutter. 

Sounds good. I think you'd be fine adding the new lockfiles in this PR as well. up to you! merge whenever.

**Comment by jomey** [comment]
Here another error from the test run:
```
Setup auth token
  Error: Action failed with "not found deploy key or tokens"
```
Log an issue?

**Comment by scottyhq** [comment]
> Error: Action failed with "not found deploy key or tokens" 

Looks like GitHub actions checks the step *inputs* even if the step does not run. I'm surprised it raises an error, if we want to avoid it, we could pass the default token for any steps that run `buildbook` but do not publish:

```
    - uses: ./.github/actions/buildbook
      with:
        jb-cache: true
        publish-to-gh: true
        jb-save: true
        token:  ${{ secrets.GITHUB_TOKEN }}
```

As in https://github.com/uwhackweek/jupyterbook-template/blob/a9d26cfa325291b5b5f1d63cea9d0085de8844d5/.github/workflows/deploy.yaml#L22

**Comment by jomey** [comment]
Alternatively we could also access the input context via `index syntax':
```
inputs['token']
```

This should not raise an error and I can see whey the 'property dereference' syntax throws an error.

https://docs.github.com/en/actions/learn-github-actions/contexts#about-contexts

**Comment by jomey** [comment]
> interesting. wouldn't using the 'index syntax' still raise an error if 'token' doesn't exist? maybe just give the 'token' input an empty string default value: `default: ' '` ?

Depends what GitHub is using for interpreting the syntax. Ruby would return a `nil` value, while Python raises a key error.
I will open a PR and test this.

**Pull request merged**